mlcomp/parallelm/pipeline/executor.py

```diff
@@ -112,10 +117,20 @@ def go(self):
             sys.stdout.flush()
 
             system_conf = pipeline[json_fields.PIPELINE_SYSTEM_CONFIG_FIELD]
-            if dag.is_stand_alone:
-                dag.run_single_component_pipeline(system_conf, self._ml_engine)
-            else:
-                dag.run_connected_pipeline(system_conf, self._ml_engine)
+            try:
+                if dag.is_stand_alone:
+                    dag.run_single_component_pipeline(system_conf, self._ml_engine)
+                else:
+                    dag.run_connected_pipeline(system_conf, self._ml_engine)
+            except SystemExit as e:
+                code = self._parse_exit_code(e.code)
+                log_message = "pipeline called exit({})".format(e.code)
+                if code != 0:
+                    self._logger.warn(log_message)
+                    raise Exception(log_message)
+                else:
+                    self._logger.info(log_message)
+                    exit(code)
```

Comment:
Do we want to use exit .. or to let the executor exit cleanly?

Reply:
Executor.go() runs in try/catch, so it exists clearly